.github/workflows/build.yml

```diff
@@ -8,31 +8,54 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: ["12.x", "14.x", "16.x"]
```

Comment:
I have several remarks related to the use of the matrix strategy:
* I don't thing single or double quotes are mandatory. And if they are necessary, it's better to favor singles quotes to stay consistent.
* Currently `upload-artifact` and `cache` steps are run for each Node.js version. I think you should use [environment variables in this matrix](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-environment-variables-in-a-matrix) combine with the use of `if` conditional on these steps ([workflow-syntax-for-github-actions#jobsjob_idstepsif](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif)).

Reply:
- [X] Double quotes for `matrix.node` removed
- [X] `if` conditional used to run upload-artifact only for one Node.js version : `12.x`